chocolatey/tools/chocolateyInstall.ps1

```diff
@@ -1,49 +1,35 @@
-
-    try {
-      if (test-path($poshgitPath)) {
-        Write-Host "Attempting to remove existing `'$poshgitPath`'."
-        remove-item $poshgitPath -recurse -force
-      }
-    } catch {
-      Write-Host "Could not remove `'$poshgitPath`'"
-    }
+﻿$ErrorActionPreference = 'Stop'
 
-    $version = "v$Env:chocolateyPackageVersion"
-    if ($version -eq 'v') { $version = 'master' }
-    $poshGitInstall = if ($env:poshGit ) { $env:poshGit } else { "https://github.com/dahlbyk/posh-git/zipball/$version" }
-    $zip = Install-ChocolateyZipPackage 'poshgit' $poshGitInstall $poshgitPath
-    $currentVersionPath = Get-ChildItem "$poshgitPath\*posh-git*\" | Sort-Object -Property LastWriteTime | Select-Object -Last 1
-
-    if(Test-Path $PROFILE) {
-        $oldProfile = @(Get-Content $PROFILE)
-
-        . $currentVersionPath\src\Utils.ps1
-        $oldProfileEncoding = Get-FileEncoding $PROFILE
-
-        $newProfile = @()
-        foreach($line in $oldProfile) {
-            if ($line -like '*PoshGitPrompt*') { continue; }
-
-            if($line -like '. *posh-git*profile.example.ps1*') {
-                $line = ". '$currentVersionPath\profile.example.ps1' choco"
-            }
-            if($line -like 'Import-Module *\src\posh-git.psd1*') {
-                $line = "Import-Module '$currentVersionPath\src\posh-git.psd1'"
-            }
-            $newProfile += $line
-        }
-        Set-Content -path $profile -value $newProfile -Force -Encoding $oldProfileEncoding
-    }
+$toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
+$moduleName = 'posh-git'  # this may be different from the package name and different case
+
+# module may already be installed outside of Chocolatey
+Remove-Module -Name $moduleName -Force -ErrorAction SilentlyContinue
+
+$sourcePath = Join-Path -Path $toolsDir -ChildPath "$modulename\*"
+$destPath = Join-Path -Path $env:ProgramFiles -ChildPath "WindowsPowerShell\Modules\$moduleName"
+
+if ($PSVersionTable.PSVersion.Major -ge 5) {
+    $manifestFile = Join-Path -Path $toolsDir -ChildPath "$moduleName\$moduleName.psd1"
+    $manifest = Test-ModuleManifest -Path $manifestFile -WarningAction Ignore -ErrorAction Stop
+    $destPath = Join-Path -Path $destPath -ChildPath $manifest.Version.ToString()
+}
+
+Write-Verbose "Creating destination directory '$destPath' for module."
+New-Item -Path $destPath -ItemType Directory -Force -ErrorAction SilentlyContinue | Out-Null
+
+Write-Verbose "Moving '$moduleName' files from '$sourcePath' to '$destPath'."
+Move-Item -Path $sourcePath -Destination $destPath -Force
 
-    $installer = Join-Path $currentVersionPath 'install.ps1'
-    & $installer
-} catch {
-  try {
-    if($oldProfile){ Set-Content -path $PROFILE -value $oldProfile -Force -Encoding $oldProfileEncoding }
-  }
-  catch {}
-  throw
+if ($PSVersionTable.PSVersion.Major -lt 4) {
+    $modulePaths = [Environment]::GetEnvironmentVariable('PSModulePath', 'Machine') -split ';'
```

Comment:
Can we assume no one will try to use Chocolatey to install this on Linux or macOS?  If not, you should use `[System.IO.Path]::PathSeparator` instead of hard-coding in `;`.

Reply:
For the time being we can. As per the next comment this is only for v0 so we don't need to worry about PS Core.